### ISSUE TITLE
Docstring issues when using construction like __doc__ = numpy.ones.__doc__

### ISSUE THREAD

**Issue by bcm-at-zama** [PyCQA/pylint#6487]
### Bug description

For file 

```python
"""Test"""
import numpy


def f():
    __doc__ = numpy.ones.__doc__
    print("doc", __doc__)


if __name__ == "__main__":
    f()
```

I get `poetry run pylint b.py`

```
b.py:6:4: W0622: Redefining built-in '__doc__' (redefined-builtin)
b.py:5:0: C0116: Missing function or method docstring (missing-function-docstring)

------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 6.67/10, +0.00)
```

while I would think that I don't do something wrong. Eg, sphinx builds my documentation correctly.

### Configuration

```ini
pylint 2.13.0, macOS 11.6
```

### Command used

```shell
poetry run pylint b.py
```

### Pylint output

```shell
b.py:6:4: W0622: Redefining built-in '__doc__' (redefined-builtin)
b.py:5:0: C0116: Missing function or method docstring (missing-function-docstring)
```

### Expected behavior

no error, I just "write" the docstring programmatically

### Pylint version

```shell
2.13.0
```

### OS / Environment

macOS 11.6

**Comment by Pierre-Sassoulas** [comment]
The warning is accurate in the sense that you're redefining a builtin, right ? If you really want to redefine builtins you can disable the check for this one file or for the whole project. Or do you think that ``__doc__`` is a special case compared to say, max / int / print and that we should filter the warning only for this builtin ?

**Comment by bcm-at-zama** [comment]
Thanks @Pierre-Sassoulas . Yes I am redefining this builtin to reuse a __doc__ from a parent, but maybe it is too hack-y. I couldn't say if whitelisting `__doc__` is appropriate, maybe I should disable the check, as you say

**Comment by bcm-at-zama** [comment]
So, up to you, we could close as "behaves correctly".

**Comment by Pierre-Sassoulas** [comment]
I'm closing as won't fix for now, if there is another argument in favor of changing something we can reexamine later.
